### PR TITLE
MMT-1049 handle orders that do not have a contact name

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: app/helpers/orders_helper.rb [hunk 1]
```diff
@@ -45,6 +45,6 @@ def contact_info(contact_address)
   end
 
   def contact_name(contact)
-    [contact['FirstName'], contact['LastName']].reject(&:empty?).join(' ')
+    [contact['FirstName'], contact['LastName']].reject(&:empty?).join(' ') unless contact.nil?
```
Comment (abbottry) [inline app/helpers/orders_helper.rb]: What is the default return value for a method? That would be returned here right? IF it's `nil` will that cause problems with whatever is calling this method? If it's just `<%= contact_name(@stuff) %>` then we'll be fine.

Comment (macrouch) [inline app/helpers/orders_helper.rb]: I can't remember what it was, but I tested it with this and it worked

```
contact = nil
[contact['FirstName'], contact['LastName']].reject(&:empty?).join(' ') unless contact.nil?
```

Comment (abbottry) [inline app/helpers/orders_helper.rb]: Within `app/views/orders/show.html.erb` the method `contact_info` passes a local variable to the 'contact_information' partial. There are a number of calls in there that use nested hash calls, so it seems like those would throw an error. Do we have a test that tests for this? I presume our tests have a contact person so we'll not see this in our tests, we should add one.

Comment (macrouch) [inline app/helpers/orders_helper.rb]: Added tests, which did reveal more problems

File: app/helpers/orders_helper.rb [hunk 2]
```diff
@@ -38,13 +38,17 @@ def order_owner(owner_guid)
...
+
     contact_address['Name'] = contact_name(contact_address)
     contact_address['Phone'] = contact_address.fetch('Phones', {}).fetch('Item', {})
 
     contact_address
   end
 
   def contact_name(contact)
-    [contact['FirstName'], contact['LastName']].reject(&:empty?).join(' ') unless contact.nil?
+    contact = {} if contact.nil?
```
Comment (abbottry) [inline app/helpers/orders_helper.rb]: The desired result here is a string 'FirstName LastName' but if nothing is supplied we're returning an empty {} -- We should match the return types. Ultimately I'm not sure the nil check is necessary?

Comment (macrouch) [inline app/helpers/orders_helper.rb]: If nothing is supplied, this method returns an empty string. I'm not sure what you mean

**Comment by abbottry** [comment]
If it's nil it returns {}, unless I'm seeing an odd diff on git.

**Comment by macrouch** [comment]
@abbottry It might be git looking weird, here is the code:

```
contact = {} if contact.nil?

[contact.fetch('FirstName', ''), contact.fetch('LastName', '')].reject(&:empty?).join(' ')
```

**Comment by abbottry** [comment]
👌🏻

**Pull request merged**